src/libraries/Microsoft.Extensions.Logging.Abstractions/src/Microsoft.Extensions.Logging.Abstractions.csproj

```diff
@@ -2,7 +2,6 @@
 
   <PropertyGroup>
     <TargetFrameworks>$(NetCoreAppCurrent);netstandard2.0;net461</TargetFrameworks>
```

Comment:
Today this configuration exists to share with ASP.NET for the shared framework.

For packages like this, how do we imagine this changing with each release?  

When `NetCoreAppCurrent` changes to `net7.0`, will this become `$(NetCoreAppCurrent);net6.0;netstandard2.0;net461`?

I think it will need to if we don't want to give folks different behavior.  Once we start removing the "special" characteristic of `NetCoreAppCurrent` then folks will no longer try to catch differences between it and other `TargetFrameworks`: that's a key benefit we're getting here.

Requiring accumulation of `TargetFrameworks` will result in some churn to the codebase, modifying conditions, ifdefs, etc.  It will make turning the crank on versioning in the repo a bit harder (hopefully tool-able).  Not saying that's a reason not to do this, but something we should be aware of and consider.  I wonder if there are other changes we can make to reduce the cost or complexity of that?  I'd like to hear some thoughts on this from @Anipik

Reply:
> When NetCoreAppCurrent changes to net7.0, will this become $(NetCoreAppCurrent);net6.0;netstandard2.0;net461?

Yes, that's the intention. For every LTS release we will carry three .NETCoreApp configurations within a package (i.e. net8.0, net7.0, net6.0) and for every non LTS release two (i.e. net6.0 and net7.0). 

> Requiring accumulation of TargetFrameworks will result in some churn to the codebase, modifying conditions, ifdefs, etc. It will make turning the crank on versioning in the repo a bit harder

I already prepared the codebases for that by not conditioning on a specific TargetFramework version in 90% of the cases. The remaining 10% of the cases will need to be updated when we target the next version of .NET. That's expected and the intentional behavior. I plan do this myself when we switch over from .NET 6 to .NET 7.

Aside from this topic, but this is also the reason why I didn't push back on your feedback to not touch previous .NETCoreApp configurations in https://github.com/dotnet/runtime/pull/53023 as this will naturally happen anyway when we target .NET 7 in the repo later this year.

Reply:
If we reach(-ed) consensus it would be great if someone would approve this PR in case I would like to depend on it for my next work item on Monday.

Reply:
> I plan do this myself when we switch over from .NET 6 to .NET 7.

Cool.  Let's make sure we keep open 👀 for any patterns when we do that, as well as any follow up PRs if we miss anything.  It's an opportunity to simplify more or document process.